src/lib/pubkey/pk_ops.cpp

```diff
@@ -31,28 +31,24 @@ size_t PK_Ops::Encryption_with_EME::max_input_bits() const {
    return 8 * m_eme->maximum_input_size(max_ptext_input_bits());
 }
 
-secure_vector<uint8_t> PK_Ops::Encryption_with_EME::encrypt(const uint8_t msg[],
-                                                            size_t msg_len,
-                                                            RandomNumberGenerator& rng) {
+std::vector<uint8_t> PK_Ops::Encryption_with_EME::encrypt(std::span<const uint8_t> msg, RandomNumberGenerator& rng) {
    const size_t max_raw = max_ptext_input_bits();
    secure_vector<uint8_t> eme_output((max_raw + 7) / 8);
-   size_t written = m_eme->pad(eme_output, std::span{msg, msg_len}, max_raw, rng);
-   return raw_encrypt(eme_output.data(), written, rng);
+   size_t written = m_eme->pad(eme_output, msg, max_raw, rng);
+   return raw_encrypt(std::span{eme_output.data(), written}, rng);
```

Comment:
Quite a nit, but I'd prefer `std::span{eme_output}.first(written)`, because its easier to parse in "natural" language.